### PR TITLE
EIP1-10513 add hasSentNotRegisteredToVoteCommunication to statistics message

### PR DESCRIPTION
**Pull request by Callum-Griffiths**
## Describe your changes
Include has sent invite to register in statistics message

## Issue ticket number and link

https://dluhcdigital.atlassian.net/browse/EIP1-10513

### CODE REVIEW

File: src/test/kotlin/uk/gov/dluhc/notificationsapi/service/SendNotificationServiceTest.kt [hunk 1]
```diff
@@ -406,7 +406,7 @@ internal class SendNotificationServiceTest {
     @ParameterizedTest
     @EnumSource(
         NotificationType::class,
-        names = ["PHOTO_RESUBMISSION", "PHOTO_RESUBMISSION_WITH_REASONS", "ID_DOCUMENT_RESUBMISSION", "ID_DOCUMENT_RESUBMISSION_WITH_REASONS", "ID_DOCUMENT_REQUIRED"],
+        names = ["PHOTO_RESUBMISSION", "PHOTO_RESUBMISSION_WITH_REASONS", "ID_DOCUMENT_RESUBMISSION", "ID_DOCUMENT_RESUBMISSION_WITH_REASONS", "ID_DOCUMENT_REQUIRED", "BESPOKE_COMM", "INVITE_TO_REGISTER"],
```
Comment (Callum-Griffiths) [inline src/test/kotlin/uk/gov/dluhc/notificationsapi/service/SendNotificationServiceTest.kt:409]: Realised this was missing BESPOKE_COMMS so added it too

File: src/test/kotlin/uk/gov/dluhc/notificationsapi/service/SendNotificationServiceTest.kt [hunk 2]
```diff
@@ -469,7 +469,7 @@ internal class SendNotificationServiceTest {
     @ParameterizedTest
     @EnumSource(
         NotificationType::class,
-        names = ["ID_DOCUMENT_RESUBMISSION", "ID_DOCUMENT_RESUBMISSION_WITH_REASONS", "ID_DOCUMENT_REQUIRED", "REQUESTED_SIGNATURE", "REJECTED_SIGNATURE", "NINO_NOT_MATCHED", "REJECTED_SIGNATURE_WITH_REASONS"],
+        names = ["ID_DOCUMENT_RESUBMISSION", "ID_DOCUMENT_RESUBMISSION_WITH_REASONS", "ID_DOCUMENT_REQUIRED", "REQUESTED_SIGNATURE", "REJECTED_SIGNATURE", "NINO_NOT_MATCHED", "REJECTED_SIGNATURE_WITH_REASONS", "BESPOKE_COMM", "INVITE_TO_REGISTER"],
```
Comment (Callum-Griffiths) [inline src/test/kotlin/uk/gov/dluhc/notificationsapi/service/SendNotificationServiceTest.kt:472]: Same here

File: src/main/kotlin/uk/gov/dluhc/notificationsapi/rest/StatisticsController.kt [hunk 3]
```diff
@@ -71,10 +74,13 @@ class StatisticsController(
 
         val bespokeCommunications = notifications.filter { it.type == NotificationType.BESPOKE_COMM }.size
 
+        val hasSentInviteToRegister = notifications.any { it.type == NotificationType.INVITE_TO_REGISTER }
+
         return CommunicationsStatisticsResponseOAVA(
```
Comment (ClaHur) [inline src/main/kotlin/uk/gov/dluhc/notificationsapi/rest/StatisticsController.kt:79]: Assume you'll have worked out how to handle this when making the bespokeComms changes before, but just making sure the service APIs will be ready to receive the new shape of CommunicationsStatisticsResponseOAVA/CommunicationsStatisticsResponseVAC without error?

File: src/main/resources/openapi/NotificationsAPIs.yaml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 openapi: 3.0.0
 info:
   title: Notifications APIs
-  version: '1.17.1'
+  version: '1.17.2'
```
Comment (ClaHur) [inline src/main/resources/openapi/NotificationsAPIs.yaml:4]: As in comment above - worth making sure this gets copied over to the service APIs so there's no errors when the response has the extra fields!

**Comment by ClaHur** [comment]
Looks good - one small clarification around backwards compatibility then happy to approve!

File: src/main/kotlin/uk/gov/dluhc/notificationsapi/rest/StatisticsController.kt (hunk 3 above)
Comment (Callum-Griffiths) [reply src/main/kotlin/uk/gov/dluhc/notificationsapi/rest/StatisticsController.kt:79]: Yep I'm working on the service API's now

Comment (Callum-Griffiths) [reply src/main/kotlin/uk/gov/dluhc/notificationsapi/rest/StatisticsController.kt:79]: Can merge this in after those if need be

Comment (ClaHur) [reply src/main/kotlin/uk/gov/dluhc/notificationsapi/rest/StatisticsController.kt:79]: yeah I don't think we ever send the CommunicationsStatisticsResponse objects from the service APIs to this one, so probably safest to merge in the service apis first once they're done

**Review by ClaHur — APPROVED** [review]

**Pull request merged**